bpf/kmesh/workload/include/authz.h

```diff
@@ -166,49 +255,146 @@ rule_match_check(Istio__Security__Rule *rule, struct xdp_info *info, struct bpf_
     return MATCHED;
 }
 
-static inline int
-do_auth(Istio__Security__Authorization *policy, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+SEC("xdp_auth")
+int policy_check(struct xdp_md *ctx)
 {
-    void *rulesPtr = NULL;
-    Istio__Security__Rule *rule = NULL;
-    int matchFlag = 0;
-    __u32 i = 0;
+    struct match_context *match_ctx;
+    wl_policies_v *policies;
+    void *rulesPtr;
+    __u32 policyId;
+    Istio__Security__Authorization *policy;
+    struct bpf_sock_tuple tuple_key = {0};
+    struct xdp_info info = {0};
+    int ret;
+
+    if (construct_tuple_key(ctx, &tuple_key, &info) != PARSER_SUCC) {
```

Comment:
instead of parsing this everywhere, can we move the parsed res into kmesh_tc_args